### PR TITLE
Break the ad headline and CTA from the body

### PR DESCRIPTION
**Pull request by davidfischer**
This breaks the ad headline, body, and CTA out of the text.

* New ads will all require setting at least the body and optionally the headline and CTA. These new fields do not support HTML.
* Existing ads will continue to support the old interface with limited HTML supported. We can either migrate those old ads or just let them phase out.
* The only tricky part of rolling this out (I think) will be publishers who already use the `body` field in our API response. They will want to also take the `headline` and `cta` fields. For old ads, `headline` and `cta` will be `None`.

Fixes: https://github.com/readthedocs/ethical-ad-server/issues/152

### CODE REVIEW

**Comment by ericholscher** [comment]
> The only tricky part of rolling this out (I think) will be publishers who already use the body field in our API response. They will want to also take the headline and cta fields. For old ads, headline and cta will be None.

This seems broken. We should just add a new field for body, and adjust the existing body copy to include the entire ad content without HTML. That is what it currently means, so we shouldn't break those integrations.

File: adserver/forms.py [hunk 1]
```diff
@@ -156,10 +163,14 @@ def clean(self):
 
             # Check text length
             if ad_type.max_text_length:
-                stripped_text = bleach.clean(text, tags=[], strip=True)
+                if text:
+                    stripped_text = bleach.clean(text, tags=[], strip=True)
+                else:
+                    stripped_text = f"{headline} {body} {cta}"
```
Comment (ericholscher) [inline adserver/forms.py:169]: This feels a bit weird. We should probably normalize the spaces to make sure it only includes 1 space.

File: adserver/models.py [hunk 2]
```diff
@@ -941,15 +974,25 @@ def offer_ad(self, request, publisher, ad_type_slug, div_id, keywords):
...
+            body = html.unescape(bleach.clean(self.text, tags=[], strip=True))
+        if not text:
+            text = self.render_links(click_url)
 
         return {
             "id": self.slug,
-            "text": self.text,
+            "text": text,
+            "headline": headline,
             "body": body,
```
Comment (ericholscher) [inline adserver/models.py:994]: Maybe we migrate the new ad data into a dict?:

```
ad:
  - headline
  - body
  - cta
```
Then keep the existing top-level body as:

```
body: bleach.clean(headline + body + cta)
```

Or maybe we rename `body` in the new ad format, so we don't have confusing overloaded terms. Perhaps `headline, content, cta`?

**Review by ericholscher — COMMENTED** [review]
This looks like a good change. I don't know if we want to block it on publishers adjusting their integration tho, especially given that it would be a weird chicken/egg problem on when they could deploy reading the new data. I think we should keep backwards compat on the API and just introduce the new data without breaking things.

File: adserver/forms.py (hunk 1 above)
Comment (davidfischer) [reply adserver/forms.py:169]: When the ad is actually rendered, this does happen correctly. This is only used to check that the ad length isn't too long.

File: adserver/models.py (hunk 2 above)
Comment (davidfischer) [reply adserver/models.py:983]: I created a new dict named `copy` and put the 3 values in there.

**Comment by davidfischer** [comment]
> This seems broken. We should just add a new field for body, and adjust the existing body copy to include the entire ad content without HTML. That is what it currently means, so we shouldn't break those integrations.

I ended up switching to what you suggested. It's a bit awkward still for folks just randomly taking `body` since it'll just be essentially "{headline} {content} {cta}" but I guess that's better than nothing.

**Review by ericholscher — APPROVED** [review]
These changes look good 👍

**Comment by davidfischer** [comment]
I have to rebase this before merging to fix the migration versions.

**Pull request merged**